crates/iota-transaction-builder/src/builder/mod.rs

```diff
@@ -0,0 +1,955 @@
+// Copyright 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Builder for Programmable Transactions.
+
+use std::{
+    collections::{BTreeMap, HashMap},
+    marker::PhantomData,
+};
+
+use iota_crypto::IotaSigner;
+use iota_graphql_client::{Client, DryRunResult};
+use iota_types::{
+    Address, GasPayment, Identifier, ObjectId, ObjectReference, Owner, ProgrammableTransaction,
+    Transaction, TransactionEffects, TransactionExpiration, TypeTag,
+};
+use serde::Serialize;
+
+use crate::{
+    builder::{
+        named_commands::{NamedCommand, NamedCommands},
+        ptb_arguments::PTBArguments,
+    },
+    error::Error,
+    publish_type::PublishType,
+    types::{MoveType, MoveTypes},
+    unresolved::{
+        Argument, Command, Input, InputId, InputKind, MakeMoveVector, MergeCoins, MoveCall,
+        Publish, SplitCoins, TransferObjects, Upgrade,
+    },
+};
+
+mod named_commands;
+pub(crate) mod ptb_arguments;
+
+/// A transaction builder which can be used to construct [`Transaction`]s.
+#[derive(Debug, Clone)]
+#[repr(C)]
+pub struct TransactionBuilder<C = (), L = ()> {
+    data: TransactionBuildData,
+    client: C,
+    last_command: PhantomData<L>,
+}
+
+/// Transaction data used to build a [`Transaction`].
+#[derive(Debug, Clone)]
+pub struct TransactionBuildData {
+    /// The inputs to the transaction.
+    inputs: BTreeMap<InputId, Input>,
+    /// The list of commands in the transaction. A command is a single operation
+    /// in a programmable transaction.
+    commands: Vec<Command>,
+    /// The gas budget for the transaction.
+    gas_budget: Option<u64>,
+    /// The gas price for the transaction.
+    gas_price: Option<u64>,
+    /// The sender of the transaction.
+    sender: Address,
+    /// The sponsor of the transaction. If None, the sender is also the sponsor.
+    sponsor: Option<Address>,
+    /// The expiration of the transaction. The default value of this type is no
+    /// expiration.
+    expiration: TransactionExpiration,
+    named_commands: HashMap<String, Argument>,
```

Comment:
Can you add a comment what this is about? It's only used to name results and pass them as argument in functions, right? Would `named_results` be a better name then?